testing/src/main/java/io/grpc/internal/testing/CensusTestUtils.java

```diff
@@ -0,0 +1,210 @@
+        }
+      }
+      return null;
+    }
+
+    public long getMetricAsLongOrFail(MetricName metricName) {
+      double doubleValue = getMetric(metricName);
+      long longValue = (long) (Math.abs(doubleValue) + 0.0001);
+      if (doubleValue < 0) {
+        longValue = -longValue;
+      }
+      return longValue;
+    }
+  }
+
+  public static final TagKey EXTRA_TAG = new TagKey("/rpc/test/extratag");
+
+  public static class FakeCensusContextFactory implements CensusContextFactory {
+    public final BlockingQueue<MetricsRecord> records = new LinkedBlockingQueue<MetricsRecord>();
+    public final BlockingQueue<FakeCensusContext> contexts =
+        new LinkedBlockingQueue<FakeCensusContext>();
+    private static final Context.Key<FakeCensusContext> CONTEXT_KEY =
+        Context.key("fakeCensusContext");
+    private final FakeCensusContext DEFAULT_CONTEXT =
+        new FakeCensusContext(ImmutableMap.<TagKey, TagValue>of(), this);
+
+    @Override
+    public CensusContext deserialize(ByteBuffer buffer) {
+      String extraTagValue = new String(buffer.array());
+      if (extraTagValue.length() > 0) {
+        return getDefault().with(EXTRA_TAG, new TagValue(extraTagValue));
+      } else {
+        return null;
+      }
+    }
+
+    @Override
+    public FakeCensusContext getCurrent() {
+      FakeCensusContext current = CONTEXT_KEY.get();
+      if (current == null) {
+        return getDefault();
+      } else {
+        return current;
+      }
+    }
+
+    @Override
+    public FakeCensusContext getDefault() {
+      return DEFAULT_CONTEXT;
+    }
+  }
+
+  public static class FakeCensusContext implements CensusContext {
+    private final ImmutableMap<TagKey, TagValue> tags;
+    private final FakeCensusContextFactory factory;
+
+    private FakeCensusContext(ImmutableMap<TagKey, TagValue> tags,
+        FakeCensusContextFactory factory) {
+      this.tags = tags;
+      this.factory = factory;
+    }
+
+    @Override
+    public Builder builder() {
+      return new FakeCensusContextBuilder(this);
+    }
+
+    @Override
+    public CensusContext with(TagKey k1, TagValue v1) {
+      return builder().set(k1, v1).build();
+    }
+
+    @Override
+    public CensusContext with(TagKey k1, TagValue v1, TagKey k2, TagValue v2) {
+      return builder().set(k1, v1).set(k2, v2).build();
+    }
+
+    @Override
+    public CensusContext with(
+        TagKey k1, TagValue v1, TagKey k2, TagValue v2, TagKey k3, TagValue v3) {
+      return builder().set(k1, v1).set(k2, v2).set(k3, v3).build();
+    }
+
+    @Override
+    public CensusContext record(MetricMap metrics) {
+      factory.records.add(new MetricsRecord(tags, metrics));
+      return this;
+    }
+
+    @Override
+    public ByteBuffer serialize() {
+      TagValue extraTagValue = tags.get(EXTRA_TAG);
+      if (extraTagValue == null) {
+        return ByteBuffer.allocate(0);
+      } else {
+        return ByteBuffer.wrap(extraTagValue.toString().getBytes());
```

Comment:
Doesn't matter. No matter what the locale is, serialize() and deserialize() are compatible.